docs/madr/decisions/035-mesh-metric-policy.md

```diff
@@ -0,0 +1,201 @@
+# MeshMetric policy
+* status: accepted
+  Technical Story: https://github.com/kumahq/kuma/issues/5708
+
+## Context and Problem statement
+We want to create a new policy responsible for Metrics collection. This policy will be part of Kuma observability policies 
+stack, alongside `MeshAccessLog` and `MeshTrace`. The new policy should expose the same functionality as now is exposed through `Mesh`. 
+We want to provide support for Prometheus and native support for OTEL backend.
+
+## Decision Outcome
+We will implement new `MeshMetric` policy as described further in this MADR.
+
+## Solution
+
+### API design
+
+#### Current API
+The API is going to be similar to what we have under `mesh.spec.metrics`. The current API looks like this:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: default
+spec:
+  metrics:
+  enabledBackend: prometheus-1
+  backends:
+    - name: prometheus-1
+      type: prometheus
+      conf:
+        tls:
+          mode: disabled | providedTLS | activeMTLSBackend
+        port: 5670
+        path: /metrics
+        tags:
+          kuma.io/service: dataplane-metrics
+        envoy:
+          filterRegex: http2_act.*
+          usedOnly: true
+        aggregate:
+          - name: my-service
+            path: "/metrics/prometheus"
+            port: 8888
+          - name: other-sidecar
+            port: 8000
+```
+
+#### Changes to the API
+1. I think the aggregation that [uses the aggregate name and pod annotations](https://kuma.io/docs/2.5.x/policies/traffic-metrics/#expose-metrics-from-applications) no longer makes sense, as we can use targetRef 
+policies to point to specific DPs. I imagine there would be a default wide policy (possibly `Mesh` scoped) and per app / framework / language overrides.
+2. What I also don’t like about this is that “filters” are under envoy - We could rename it to `sidecar` and introduce 
+the concept of “profiles” like we discussed. Maybe something like this:
+
+```yaml
+sidecar:
+  usedOnly: true # true or false
+  profile: minimal # one of minimal, default, full
+  regex: http2_act.* # only profile or regex can be defined
```

Comment:
Just an idea, not sure if useful. Sometimes you want to get `http2_act` and maybe `cx_active` for this you need to create some more complicated regex (if works) but you could maybe do 2 separate call. I feel like doing x separate calls is not the best option but we could verify if more complicated regexes works.

Reply:
I just checked and more complicated regexes with `or` works, for example: `/stats?filter=circuit_breakers|update_success` 

In this mechanism, we are utilizing envoy filtering functionality, and I don't think making additional calls it good way to do it. In the future with dynamic dp config and our producers we can add more advanced filtering that will be easier for user to configure